Geopal import

azime opened the pull request.

azime committed:
ed : index added

azime committed:
ed : modify "boundary" to MULTIPOLYGON type

azime committed:
ed : modify "boundary" to MULTIPOLYGON type

azime committed:
ed : Ajout des fonctions

1) conversion des coordonnées en WGS84
2) calcul des limites des admins

azime committed:
ed : modification du type de POLYGON en multipolygon de BOUNDARY

azime committed:
ed : import des données GEOPAL

kinnou02 commented:
Une description serait la bienvenue

azime committed:
ed : geopal2ed

kinnou02 commented on source/ed/connectors/geopal_parser.h:
```diff
@@ -0,0 +1,35 @@
+#include <utils/logger.h>
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string.hpp>
+//#include "type/type.h"
```
si on en a pas besoin, autant les supprimer...

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+namespace ed { namespace connectors {
+
+GeopalParser::GeopalParser(const std::string& path): path(path){
+    init_logger();
```
Ce n'est pas au parser geopal d'initialiser le logger

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+
+void GeopalParser::fill_nodes(ed::Georef& data){
+    for(const std::string file_name : this->files){
+        std::string adresse(file_name.begin(), file_name.begin() + 7);
```
aussi connu sous le nom de [start_with](http://www.boost.org/doc/libs/1_55_0/doc/html/boost/algorithm/starts_with.html)

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+                uri = row[x_c] + row[y_c];
+                auto adm = data.admins.find(row[insee_c]);
+                if (adm != data.admins.end()){
+                    if(reader.is_valid(code_post_c, row))
```
Je déteste qu'on ne mette pas les accolades même si elles sont facultatives

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+            if (reader.is_valid(x1, row) && reader.is_valid(y1, row)
+                && reader.is_valid(x2, row) && reader.is_valid(y2, row)
+                && reader.is_valid(inseecom_d, row)){
+                    auto admin = data.admins.find(row[inseecom_d]);
```
Le niveau d'indentation n'est pas bon

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+                            data.edges[edge_uri]= edg;
+                        }
+                   }
+                }
```
il y a 3 espace au lieu de 4

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+                           target_node = target_it->second;
+                        }
+                        std::hash<std::string> hash_fn;
+                        ed::types::Way* current_way;
```
toujours initialiser un pointeur à nullptr si il n'est pas initialiser imédiatement

azime committed:
ed : corrections suite aux remarques

azime commented:
corrections suite aux remarques

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+        }
+    }
+}
+}
```
toujours spécifier que c'est une accolade de fin de namespace, vu qu'on ne les indentes pas

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+        if(!boost::algorithm::starts_with(file_name, "adresse")){
+            continue;
+        }
+        CsvReader reader(this->path + "/" + file_name, ';', true, true);
```
Ca ne compile pas chez moi
ok, j'ai rien dit j'avais pas vu la derniére mise à jours du  module csv

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+                auto adm = data.admins.find(row[insee_c]);
+                if (adm != data.admins.end()){
+                    if(reader.is_valid(code_post_c, row))
+                        adm->second->postcode = row[insee_c];
```
mauvaise colonne insee != postcode

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+                 if (iter->path().extension() == ".txt"){
+                    std::string file_name = iter->path().filename().string();
+                    boost::to_lower(file_name);
+                     this->files.push_back(file_name);
```
espace

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+            LOG4CPLUS_WARN(logger, "Immpossible d'ouvrir le fichier " + reader.filename);
+            return;
+        }
+        std::vector<std::string> mandatory_headers = {"numero", "code_insee", "code_post", "x_adresse", "y_adresse"};
```
numéro à pas l'air très obligatoire vu qu'on ne s'en sert pas

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+            return;
+        }
+        std::vector<std::string> mandatory_headers = {"x_debut" , "y_debut", "x_fin", "y_fin", "longueur", "inseecom_g",
+        "inseecom_d", "bornedeb_d", "bornedeb_g", "bornefin_d", "bornefin_g"};
```
vu qu'on ne se sert jamais des bornes, elles ne sont pas obligatoires

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+                    }
+                    std::hash<std::string> hash_fn;
+                    ed::types::Way* current_way = nullptr;
+                    std::string wayd_uri = std::to_string(hash_fn(row[nom_voie_d])) + row[inseecom_d];
```
ça suppose que dans une même ville on a jamais deux rues différentes avec le même nom.
ça fera l'affaire pour le moment mais faut au moins mettre un commentaire pour le tracer + un jira pour amélioration

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+                        wy->admin = admin->second;
+                        admin->second->is_used = true;
+                        wy->id = data.ways.size() + 1;
+                        if(reader.is_valid(nom_voie_d, row))
```
indentation + accolade

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+                        edg->source->is_used = true;
+                        edg->target = target_node;
+                        edg->target->is_used = true;
+                        if(reader.is_valid(l, row))
```
indentation + accolade

kinnou02 commented on source/ed/connectors/geopal_parser.cpp:
```diff
@@ -0,0 +1,254 @@
+                        if (hn == data.house_numbers.end()){
+                            ed::types::HouseNumber* current_hn = new ed::types::HouseNumber;
+                            current_hn->coord = navitia::type::GeographicalCoord(str_to_double(row[x_c]), str_to_double(row[y_c]));
+                            current_hn->number = str_to_int(row[numero_c]);
```
donc number

kinnou02 commented on source/ed/geopal2ed.cpp:
```diff
@@ -0,0 +1,76 @@
+    geopal_parser.fill(data);
+
+//    ed::connectors::FusioParser fusio_parser(input);
+//    fusio_parser.fill(data, date);
```
Pourquoi laisser le code réalisant l'appel de fusio2ed

kinnou02 commented on source/ed/geopal2ed.cpp:
```diff
@@ -0,0 +1,76 @@
+    p.persist(data);
+    std::cout<<std::endl<<"temps :"<<(pt::microsec_clock::local_time() - start).total_milliseconds()<<std::endl;
+
+    return 0;
```
Il n'y a strictement aucune gestion d'erreur, soit on plante salement, soit on retourne 0.

kinnou02 replied:
Il faudra donc rajouter un catch de std::exception juste pour logger l'erreur, quitte à la re-trow pour la stacktrace

kinnou02 commented on source/ed/types.h:
```diff
@@ -292,4 +292,49 @@ struct StopTime : public Nameable {
+    size_t id;
+    bool is_used;
+    navitia::type::GeographicalCoord coord;
+    Node():is_used(false){};
```
id n'est aps initialisé

kinnou02 commented on source/ed/types.h:
```diff
@@ -292,4 +292,49 @@ struct StopTime : public Nameable {
+    std::string name;
+    std::string postcode;
+    navitia::type::GeographicalCoord coord;
+    Admin():is_used(false), level("8"){};
```
id n'est pas initialisé

kinnou02 commented on source/ed/types.h:
```diff
@@ -292,4 +292,49 @@ struct StopTime : public Nameable {
+    Node* source;
+    Node* target;
+    int length;
+    Edge():way(nullptr), source(nullptr), target(nullptr){}
```
length n'est pas initialisé

kinnou02 commented on source/ed/types.h:
```diff
@@ -292,4 +292,49 @@ struct StopTime : public Nameable {
+    Way* way;
+    navitia::type::GeographicalCoord coord;
+
+    HouseNumber(): number(-1), way(nullptr){}
```
Si number représente le numéro de la rue, alors un int n'est pas utilisable, exemple "25 bis"

azime replied:
oui, il faut changer le MCD

kinnou02 replied:
pourquoi?
          Table "georef.house_number"
  Column   |         Type          | Modifiers 
-----------+-----------------------+-----------
 way_id    | bigint                | 
 coord     | geography(Point,4326) | not null
 number    | text                  | not null
 left_side | boolean               | not nul

kinnou02 commented on source/sql/ed/34-functions.sql:
```diff
@@ -186,3 +186,40 @@ and ad2.level > ad1.level
+ -- Conversion des coordonnées vers wgs84
+CREATE OR REPLACE FUNCTION georef.coord2wgs84(lon real, lat real, coord_in int)
+  RETURNS RECORD AS
+$func$
```
pourquoi ne pas avoir utiliser "$$" comme pour toutes les autres?

kinnou02 commented on source/sql/ed/34-functions.sql:
```diff
@@ -186,3 +186,40 @@ and ad2.level > ad1.level
+BEGIN
+ SELECT ST_X(aa.new_coord::geometry) as lon, ST_Y(aa.new_coord::geometry) as lat from (
+ SELECT ST_Transform(ST_GeomFromText('POINT('||lon||' '||lat||')',coord_in),4326) As new_coord)aa INTO ret;
+RETURN ret;
```
la barre d'espace et de tabulations font gréve? :)

kinnou02 commented on source/ed/ed_persistor.cpp:
```diff
@@ -4,6 +4,139 @@
+void EdPersistor::persist(const ed::Georef& data){
+
+    this->lotus.start_transaction();
+    std::cout << "début : vider toutes les tables (TRUNCATE)!" << std::endl;
```
il faut privilégier l'utilisation du logger

kinnou02 commented on source/ed/ed_persistor.cpp:
```diff
@@ -4,6 +4,139 @@
+    navitia::type::GeographicalCoord to_return;
+    PGresult* res = nullptr;
+    try{
+        res = this->lotus_second.exec("select lon, lat from georef.coord2wgs84(" + std::to_string(coord.lon()) + "," + std::to_string(coord.lat()) + "," + std::to_string(coord_in)
```
ya peut etre plus efficace pour convertir des coordonnées

kinnou02 commented on source/ed/ed_persistor.cpp:
```diff
@@ -4,6 +4,139 @@
+        std::string lat = PQgetvalue(res, 0, 1);
+        to_return = navitia::type::GeographicalCoord(str_to_double(lon), str_to_double(lat));
+        PQclear(res);
+    }catch(...){
```
JAMAIS de catch(...) ><
surtout que libpq ne va jamais throw d'exception, vu que c'est du C, donc il n'y a que lotus qui peut throw

kinnou02 commented on source/ed/ed_persistor.cpp:
```diff
@@ -4,6 +4,139 @@
+    geog << std::cout.precision(10);
+        for(auto itm : data.admins){
+            if(itm.second->is_used){
+            navitia::type::GeographicalCoord coord = this->coord2wgs84(itm.second->coord);
```
indentation

kinnou02 commented on source/ed/ed_persistor.cpp:
```diff
@@ -4,6 +4,139 @@
+            geog <<"POINT("<<coord.lon()<<" "<<coord.lat()<<")";
+            this->lotus.insert({std::to_string(itm.second->id), itm.second->name,
+                               itm.second->postcode, itm.second->insee,
+                               itm.second->level, geog.str() , itm.first});
```
ça serait plus clair d'utiliser directement le code insee

kinnou02 commented on source/ed/ed_persistor.cpp:
```diff
@@ -4,6 +4,139 @@
+
+void EdPersistor::insert_ways(const ed::Georef& data){
+    this->lotus.prepare_bulk_insert("georef.way", {"id", "name", "uri", "type"});
+    for(auto itm : data.ways){
```
vu que je suis parano j'aurais tendance à forcer l'utilisation d'une référence constante

kinnou02 commented:
accessoirement quasi tous les logs, commentaires et messages de commit sont en français

kinnou02 commented:
Et il faudra rebase sur dev avant de pouvoir merger

azime committed:
ed : ajout de la gestion d'exception

azime committed:
ed : ecriture des logs, gestion des exceptions

azime committed:
ed:  corrections suites aux remarques

Ajout des exceptions

azime commented:
correction suite aux remarques

azime committed:
ed : Ajout d'un paramètre "percent_delete"

Pour supprimer les graphes non connectées
avec un pourcentage < à "percent_delete"

azime committed:
ed : procedure de migration

migration de la colonne boundary

azime commented:
modification de la procédure de migration de la colonne BOUNDARY de polygon en multipolygon

azime committed:
ed : modification du type number_house

azime commented:
modification du type house_number de int en text

azime committed:
ed : Amélioration de la fonction de

calcul des limites des communes

azime commented:
amélioration de la fonction de calcul des limites des communes
